openapi.yaml

```diff
@@ -3268,19 +3269,22 @@ paths:
         The result will be stored in the format specified in the process.
         To specify the format use a process such as `save_result`.
 
-        This endpoint has no effect if the job status is already `queued` or
-        `running`. In particular, it doesn't restart a running job. Processing
-        MUST be canceled before to restart it.
-
         The job status is set to `queued`, if processing doesn't start
-        instantly.
+        instantly. The same applies if the job status is `canceled`, `finished`,
+        or `error`, which restarts the job and discards previous results.
+        Clients SHOULD warn users and ask for confirmation if results may get
```

Comment:
I'm not a fan of moving this responsibility to the client (library). The client is then forced to get the status before sending the start instruction and there must be a way to interactively ask the user for confirmation (which is not straightforward in scripting/headless contexts). Due to latency there is also risk on race conditions.

As mentioned elsewhere I would propose that a user first has to actively cancel a finished job before it can be restarted (and a "start" on a finished job does nothing or raises error)

Reply:
It's not really doable in a compliant way, e.g. right now the DELETE endpoint for .../results says it has no effect for states other than queued or running. In the future we could make this endpoint a general cancel + delete endpoint, indeed. So I'd propose to open an issue for a 2.x version?

Reply:
ok I understand that the feature "cancel before restart" requires v2-style changes.

But here we are defining the behavior what happens when sending "start" action when status is finished/canceled/error.

This PR explicitly adds destructive behavior for the "finished" case, while it previously was not destructive if you extrapolate from:
> This endpoint has no effect if the job status is already `queued` or
        `running`. In particular, it doesn't restart a running job. Processing
        MUST be canceled before to restart it.

I know it doesn't mention the "finished", but it sets a sentiment to handle batch job results defensively: starting a job should not break/remove what was already achieved.

I'd prefer to make this defensive behavior more explicit (e.g. no effect, or throw error) instead of the destructive handling for status "finished".

I'm fine with handling start on canceled/error status as proposed in PR.

Reply:
I don't understand where this PR adds destructive behavior? This is simply meant to clarify existing behavior. 'finished' wasn't mentioned intentionally and as such, it was destructive before this PR already.

Reply:
> 'finished' wasn't mentioned intentionally and as such, it was destructive before this PR already.

I don't completely agree: the behavior was undocumented, so it was up to the back-end implementation what actual behavior you would get.  For example with the VITO back-end, since https://github.com/Open-EO/openeo-python-driver/commit/cf1a2036d6b0e8d2b5ad99fb98bacae34ec60c41  (Feb 2022), a start on a finished job does nothing. Before that change, it was indeed destructive (with chance of having some left-over results from the first run).  I can imagine that other back-ends had different behavior (e.g. 500 Internal error for some reason)